service/qos/service_level_controller.cc

```diff
@@ -402,6 +402,11 @@ future<> service_level_controller::do_add_service_level(sstring name, service_le
     return make_ready_future();
 }
 
+bool service_level_controller::is_v2() const {
+    //FIXME: return actual state (fixed with migration commit)
+    return true;
+}
```

Comment:
I'm ok with both approaches. In general I tend to play safe in such cases, i.e. two feature checks would allow to craft 1 line emergency patch which would disable faulty feature.

As for whether auth-v2 works without qos-v2, it does. The other way around not yet sure - but probably as well.

In theory we should be able to remove whole auth-v1 with next minor release (when we're sure it's no longer used) because migration code doesn't depend on any of auth-v1 code, but dunno yet what'll be decided - potential drawback is doing backports and not allowing to easily skip versions when upgrading.

Reply:
I've stayed with separate v2 checks, but I don't have a strong opinion which option is strictly better. 
If you think I should use `auth::legacy_mode`, let me know

Reply:
Ok, let's continue with the safe option, i.e. have separate checks. After release we will find out whether they landed in the same release, and if they do, it shouldn't be difficult to simplify in a follow up.